src/internal.c

```diff
@@ -16110,9 +16110,31 @@ int ProcessReply(WOLFSSL* ssl)
                                       in->buffer + in->idx,
                                       ssl->curSize - (word16)digestSz);
                         if (ret == 0) {
-                            ssl->keys.padSz =
-                                    in->buffer[in->idx + ssl->curSize -
-                                               digestSz - 1];
+                            byte invalid = 0;
+                            byte padding = (byte)-1;
+                            word32 i;
+                            word32 off = in->idx + ssl->curSize - digestSz - 1;
+
+                            /* Last of padding bytes - indicates length. */
+                            ssl->keys.padSz = in->buffer[off];
+                            /* Constant time checking of padding - don't leak
+                             * the length of the data.
+                             */
+                            /* Compare max pad bytes or at most data + pad. */
+                            for (i = 1; i < MAX_PAD_SIZE && off >= i; i++) {
+                                /* Mask on indicates this is expected to be a
+                                 * padding byte.
+                                 */
+                                padding &= ctMaskLTE(i, ssl->keys.padSz);
+                                /* When this is a padding byte and not equal
+                                 * to length then mask is set.
+                                 */
+                                invalid |= padding &
+                                           ctMaskNotEq(in->buffer[off - i],
+                                                       ssl->keys.padSz);
+                            }
+                            /* If mask is set then there was an error. */
+                            ret = ((int)(char)invalid) & DECRYPT_ERROR;
```

Comment:
Is `&` the right binary operation here? Seems odd to AND the `DECRYPT_ERROR` with some set bits. Maybe `ret = invalid ? DECRYPT_ERROR : 0`?

Reply:
It is all part of the constant time stuff, but at this point I don't see It mattering.
Changed to an if statement.